### PR TITLE
manifest: Sync against Sony's 10.0.0_r41 on 20210421

### PR DESCRIPTION
**Pull request by Thaodan**
Sync against Sony's 10.0.0_r41 for updated device and
vendor repositories including 5GHz WiFi signal strength fix.

### CODE REVIEW

**Comment by mlehtima** [comment]
We should update all repos to latest revisions matching Sony repos. There are for example changes to audio hal and other places.

**Comment by krnlyng** [comment]
lgtm but before we can merge this we need one extra PR to droid-src-sony to remove these patches: https://github.com/mer-hybris/droid-src-sony/tree/hybris-sony-aosp-10/patches/vendor/qcom/opensource/audio-hal/primary-hal

File: tagged-localbuild.xml [hunk 1]
```diff
@@ -1,731 +1,742 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <manifest>
-  <remote fetch="https://github.com/NXPNFCProject/" name="NXP"/>
-  <remote fetch="https://android.googlesource.com/" name="aosp"
-          review="https://android-review.googlesource.com/"/>
-  <remote fetch="https://github.com/exfatprogs/" name="exfatprogs"/>
-  <remote fetch="https://github.com/sonyxperiadev/" name="sony"/>
+  <remote name="NXP" fetch="https://github.com/NXPNFCProject/"/>
+  <remote name="aosp" fetch=".." review="https://android-review.googlesource.com/"/>
```
Comment (sledges) [inline tagged-localbuild.xml]: fetch= should be as in the previous manifest, otherwise it will look for AOSP in github.com/ :)

**Review by sledges — CHANGES_REQUESTED** [review]

File: tagged-localbuild.xml (hunk 1 above)
Comment (Thaodan) [reply tagged-localbuild.xml]: So repo does not check that? nice.

**Comment by Thaodan** [comment]
> lgtm but before we can merge this we need one extra PR to droid-src-sony to remove these patches: https://github.com/mer-hybris/droid-src-sony/tree/hybris-sony-aosp-10/patches/vendor/qcom/opensource/audio-hal/primary-hal

That is included already.

File: tagged-localbuild.xml (hunk 1 above)
Comment (sledges) [reply tagged-localbuild.xml]: repo just executes everything you specify in manifest. This adds flexibility to be able to do things like `repo init -u ../parentdir-localtest`

**Review by sledges — CHANGES_REQUESTED** [review]
`localbuild.xml` is now out of sync for the newly added sm8150 repos

**Comment by sledges** [comment]
Commit message structure is seen better as:
```
foo: bar

description

[foo] bar bug#ref

(signed-off etc)
```

**Comment by Thaodan** [comment]
> `localbuild.xml` is now out of sync for the newly added sm8150 repos

Thanks for pointing that out, I added those.

**Comment by sledges** [comment]
I've switched to the new repo manifest format fields order and cherry-picked your commit (with corrected message) on top to see changes:
https://github.com/sledges/android/commits/5ghz-update

`vi` sequence for the changes to the old manifest for future ref:
```
%s/project clone-depth="1" \(.*\)\/>/project \1 clone-depth="1"\/>/g
%s/project groups="\(.*\)" name=\(.*\) clone-depth="1"\/>/project name=\2 groups="\1" clone-depth="1"\/>/g
%s/project groups="\(.*\)" name=\(.*\)\/>/project name=\2 groups="\1"\/>/g
%s/project groups="\(.*\)" name=\(.*\)>/project name=\2 groups="\1">/g
%s/dest="\(.*\)" src="\(.*\)"\/>/src="\2" dest="\1"\/>/g
%s/fetch="\(.*\)" name="\(.*\)"/name="\2" fetch="\1"/g
%s/enabled-list="\(.*\)" in-project="\(.*\)"/in-project="\2" enabled-list="\1"/g
```

**Comment by krnlyng** [comment]
> > lgtm but before we can merge this we need one extra PR to droid-src-sony to remove these patches: https://github.com/mer-hybris/droid-src-sony/tree/hybris-sony-aosp-10/patches/vendor/qcom/opensource/audio-hal/primary-hal
> 
> That is included already.

Ah i can see it now. Usually we point the manifest to the merge commit not the commit from the PR, i think we should do the same here i.e. merge the droid-src-sony PR, and use the merge commit revision here.

**Comment by sledges** [comment]
> Ah i can see it now. Usually we point the manifest to the merge commit not the commit from the PR, i think we should do the same here i.e. merge the droid-src-sony PR, and use the merge commit revision here.

I'm pretty certain that's what he meant to do after that other PR gets merged:) Otherwise every forced-push to that PR would need an update in this one

**Comment by mlehtima** [comment]
Please update device-sony-seine hash to match latest upstream revision. There were new changes added which help with echo cancellation.

**Review by mlehtima — APPROVED** [review]

**Review by mkosola — APPROVED** [review]

**Review by sledges — APPROVED** [review]

**Pull request merged**